### PR TITLE
User guide: explain root device hints

### PR DESCRIPTION
**Pull request by dtantsur**
Add plenty of links to the new document since it's quite important and
often overlooked. Explain how the deprecated profiles are related to it.
Provide an example of using HardwareData (another often overlooked
feature).

Also sneak in a couple of fixes for invalid relative URLs (sorry!).

Signed-off-by: Dmitry Tantsur <dtantsur@protonmail.com>

### CODE REVIEW

**Review by kashifest — COMMENTED** [review]
/approve

File: docs/user-guide/src/bmo/root_device_hints.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+# Root Device Hints
+
+Bare-metal machines often have more than one block device, and in many cases
```
Comment (tuminoid) [inline docs/user-guide/src/bmo/root_device_hints.md:3]: One day we will unify how we write bare-metal, baremetal or bare metal, but not today :)

File: docs/user-guide/src/bmo/root_device_hints.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    "rotational": true,
+    "serialNumber": "drive-scsi0-0-0-0",
+    "sizeBytes": 32212254720,
+    "type": "HDD",
+    "vendor": "QEMU"
+  }
+]
+```
+
+Here you can find all necessary fields to populate the root device hints before
```
Comment (tuminoid) [inline docs/user-guide/src/bmo/root_device_hints.md:98]:
```suggestion
Here you can find all necessary fields to populate the root device hints before
```
Here is typically used to provide a link or following output, but now its after the output, so here is weird to me.  Could this string be placed above the output?

**Review by tuminoid — COMMENTED** [review]
A nit.

Thanks for contributing many valuable documentation improvements @dtantsur !

File: docs/user-guide/src/bmo/root_device_hints.md (hunk 2 above)
Comment (dtantsur) [reply docs/user-guide/src/bmo/root_device_hints.md:98]: Let me see

File: docs/user-guide/src/bmo/root_device_hints.md (hunk 1 above)
Comment (dtantsur) [reply docs/user-guide/src/bmo/root_device_hints.md:3]: I think, and wikipedia seems to be consistent with it in https://en.wikipedia.org/wiki/Bare_machine, we need to use bare-metal as an auxiliary word and "bare metal" as a separate concept.

Comment (jayofdoom) [reply docs/user-guide/src/bmo/root_device_hints.md:3]: FWIW, it appears in OpenStack, we're officially "Bare Metal" for Ironic. I don't think we use that consistently though.

Comment (tuminoid) [reply docs/user-guide/src/bmo/root_device_hints.md:3]: I'll resolve this for now, thanks for taking a look and providing input @dtantsur @jayofdoom

**Pull request merged**